src/Type/Laminas/PluginMethodDynamicReturnTypeExtension/AbstractPluginMethodDynamicReturnTypeExtension.php

```diff
@@ -36,18 +39,25 @@ abstract class AbstractPluginMethodDynamicReturnTypeExtension implements Dynamic
         MethodCall $methodCall,
         Scope $scope
     ): Type {
-        $pluginManager = $this->serviceManagerLoader->getServiceLocator($this->getPluginManagerName());
-        $argType       = $scope->getType($methodCall->args[0]->value);
-        if (! $argType instanceof ConstantStringType) {
-            throw new \PHPStan\ShouldNotHappenException(\sprintf('Argument passed to %s::%s should be a string, %s given',
-                $methodReflection->getDeclaringClass()->getName(),
-                $methodReflection->getName(),
-                $argType->describe(VerbosityLevel::typeOnly())
-            ));
+        $argType = $scope->getType($methodCall->args[0]->value);
+        $strings = TypeUtils::getConstantStrings($argType);
+        $plugin = count($strings) === 1 ? $strings[0]->getValue() : null;
+        
+        if ($plugin) {
+            $pluginManager = $this->serviceManagerLoader->getServiceLocator($this->getPluginManagerName());
+            return new ObjectType(\get_class($pluginManager->get($plugin)));
         }
-        $pluginClass   = \get_class($pluginManager->get($argType->getValue()));
 
-        return new ObjectType($pluginClass);
+        if ($argType instanceof StringType) {            
```

Comment:
Is there a way in this case to get the string raw value?

Reply:
If the type is `ConstantStringType` then you can ask for `$argType->getValue()`